talk/expert.tex

```diff
@@ -403,11 +403,45 @@
       struct hasFoo<T, decltype(std::declval<T>().foo())>
         : std::true_type {};
 
-      std::cout << hasFoo<MyType>::value << std::endl;
+      struct A{}; struct B{void foo();}; struct C{int foo();};
+      static_assert(!hasFoo<A>::value, "A has no Foo()");
+      static_assert(hasFoo<B>::value, "B has Foo()");
+      // static_assert(hasFoo<C>::value, "C has Foo()"); fails assertion!
     \end{cppcode*}
   \end{exampleblock}  
 \end{frame}
 
+\begin{frame}[fragile]
+  \frametitleii{Using \texttt{void\_t} to simplify SFINAE expressions \hfill \cpp17}
+  \begin{block}{Concept}
+    \begin{cppcode*}{gobble=2}
+      template <typename... >
+      using void_t = void;
+    \end{cppcode*}
+
+    \begin{itemize}
+    \item Maps a sequence of given types to void
+    \item Introduced in \cpp17 though trivial to copy to \cpp11
+    \item Can thus be used in specialization to check the validity of an expression
+    \end{itemize}
+  \end{block}
+  \begin{exampleblock}{Example}
+    \begin{cppcode*}{}
+      template <typename T, typename = void>
+      struct hasFoo : std::false_type {};
+
+      template <typename T>
+      struct hasFoo<T, std::void_t<decltype(std::declval<T>().foo())>>
+      : std::true_type {};
+
+      struct A{}; struct B{void foo()}; struct C{int foo();};
+      static_assert(!hasFoo<A>::value, "A has no Foo()");
+      static_assert(hasFoo<B>::value, "B has Foo()");
+      static_assert(hasFoo<C>::value, "C has Foo()");
+    \end{cppcode*}
+  \end{exampleblock}
+\end{frame}
```

Comment:
Again, does not fit in the slide. And by far this time, things are cut after line 434.
Now first 2 lines can be put on 1, last bullet could be changed to "Used to ..." to be on one line and then you can put hte code in `\small` font. Not sure it will be enough though